CONTRIBUTING.md

```diff
@@ -164,6 +166,80 @@ Provide rationale; the more the better. We aim to explain design and code decisi
 
 -------------------------------------------------------------------------------------------------
 
+### Code Standard
+
+As mentioned previously, we try to make our code very easy to understand and readable for the human user. Here are some do's and don'ts
+
+#### Describe the function
+
+Make sure to include docs that gives details on the parameters and returns of the function. The name of the function should help make clear the intention of the function. This standard applies for any SASS, JS, JQuery or React JS functions.
+
+Don't:
+
+```js
+function CalculateSpecs(initialSize,endSize,speed){
+  ...
+}
+```
+
+Do
+
+```js
+* @param  {integer} initialSize - The initial size of the element to animate
+* @param  {integer} endSize     - The size the element after the animation completes
+* @param  {string}  speed       - The speed of the animation in ms
+*
+* @return {object}              - Required steps, stepSize and intervalTime for the animation
+*/
+function CalculateAnimationSpecs( initialSize, endSize, speed ) {
+  ...
+}
+
+```
+
+#### Space out the code
+
+Add space when there is code between round, square or curly brackets.
+
+Don't:
+
+```scss
+$isBlackBgContrast:AU-color-contrast(#000,$background,true,false)>=$ratio;
+
+```
+
+Do:
+
+```scss
+$isBlackBgContrast: AU-color-contrast( #000, $background, true, false ) >= $ratio;
+```
+
+#### Use new lines
```

Comment:
1. Mix concatenation and template string is quiet confusing in understanding which style you prefer. You are using and not using the power of template string in same time. And why here template string ```${ dark ? ` au-control-input--dark` : '' }```, but here ```${ status === 'valid' ? ' au-control-input--valid' : '' }``` - not. 
2. Why this line  `au-control-input ${ className }`  not splitted in 
```
'au-control-input'  +
` ${className}` + ...
```
And if staring to split - split all. And should clarify the limit of allowed length of one line string.

```jsx
// Good
<div prop={oneProp} longPropName={sameLongPropName} className={`classList`} >

// Bad, to long

<div prop={oneProp} longPropName={sameLongPropName} someVeryLongPropNameforSomeReason={propValue} >

// Should use new lines

<div
 prop={oneProp}
 longPropName={sameLongPropName}
 someVeryLongPropNameforSomeReason={propValue} 
>

// Don't

<div
 prop={oneProp} longPropName={sameLongPropName}
 someVeryLongPropNameforSomeReason={propValue} 
>
```
3. In that example in think it's better to recommend not to calculate class name with many conditions in template. It's hard to read. Extract calculations and use array so you don't need to worry about space before class name and mix template stings and concatenation.

```jsx
// to much calculations in template
<div className={
  `au-control-input ${ className }` +
  `${ dark ? ` au-control-input--dark` : '' }` +
  `${ alt ? ` au-control-input--alt` : '' }` +
  `${ small ? ` au-control-input--small` : '' }` +
  `${ block ? ` au-control-input--block` : '' }` +
  `${ status === 'valid' ? ' au-control-input--valid' : '' }` +
  `${ status === 'invalid' ? ' au-control-input--invalid' : '' }`
} 
  otherProp={propValue}
>

// extract calculations

const classes = [
		'au-control-input',
		className,
		dark ? 'au-control-input--dark' : '',
		alt ? 'au-control-input--alt' : '',
		small ? 'au-control-input--small' : '',
		block ? 'au-control-input--block' : '',
		status === 'valid' ? 'au-control-input--valid' : '',
		status === 'invalid' ? 'au-control-input--invalid' : ''
	]
		.filter( cls => cls !== '' )
		.join(' ');

<div className={classes} otherProp={propValue} >

```

Reply:
Hey @toxamiz,

Thanks for the feedback! I think a lot of this could be fixed if we can resolve #429. I've seen you've opened up a PR so thanks for that!

1.  Yeah this is probably just a bit of inconsistency. The code base is a couple of years old. We aren't perfect by a long stretch 😅. This is why we created #429 and are updating the contributing guidelines. This is for everyone 🙂

2. I agree 100%.

3. Hmmm I think the way it's done now is cleaner and easier to read. I guess that's subjective. @adamzerella thoughts?

Reply:
I agree with @sukhrajghuman here, option 1 reads nicer to me and doesn't require calling two additional functions

Reply:
> Hmmm I think the way it's done now is cleaner and easier to read. I guess that's subjective.

Yes, it is subjective :)

I prefer to keep template clear. I want to see the result of calculation, not how they been calculated. Except some simple operations like comparison and single ternary operator. If i want to know calculation process i would go to related method/function. 

something like this:

```jsx
// Option 1
// So what do we have
<div className={ // classname caclulation
  `au-control-input ${ className }` + // ok
  `${ dark ? ` au-control-input--dark` : '' }` + // ok 
  `${ alt ? ` au-control-input--alt` : '' }` + // ok
  `${ small ? ` au-control-input--small` : '' }` + // ok
  `${ block ? ` au-control-input--block` : '' }` + // ok
  `${ status === 'valid' ? ' au-control-input--valid' : '' }` + // ok
  `${ status === 'invalid' ? ' au-control-input--invalid' : '' }` // ok
} 
  someVeryImportantProp={propValue} // aha, that what i need
>

// Option 2
// Some classname calculation. Not interested, skip
const classes = [
		'au-control-input',
		className,
		dark ? 'au-control-input--dark' : '',
		alt ? 'au-control-input--alt' : '',
		small ? 'au-control-input--small' : '',
		block ? 'au-control-input--block' : '',
		status === 'valid' ? 'au-control-input--valid' : '',
		status === 'invalid' ? 'au-control-input--invalid' : ''
	]
		.filter( cls => cls !== '' )
		.join(' ');

// 
<div className={classes} somsomeVeryImportantProp={propValue} >  // aha, that what i need

```

Reply:
> Yes, it is subjective :)

It is, but we can look for evidence among industry when we are trying to come to a census. For example, AirBnB prefer template literals in their style guide: 

> Why? Template strings give you a readable, concise syntax with proper newlines and string interpolation features.

Source: http://airbnb.io/javascript/

Reply:
@adamzerella I was talking about extracting logic from template, not about using concatenation via array.join instead of template strings.

```jsx
const classes = `au-control-input ${ className } { dark ? 'au-conlatrol-input--dark' : '' } ${ alt ? 'au-control-input--alt' : '' } ${ small ? 'au-control-input--small' : '' } ${ block ? 'au-control-input--block' : '' } ${ status === 'valid' ? 'au-control-input--valid' : '' } ${ status === 'invalid' ? 'au-control-input--invalid' : '' }`

// or better

const classes = `au-control-input` +
 (className ? ` ${className}` : ``) +
 (dark ? ` au-conlatrol-input--dark` : `` ) + 
 (small ? ` au-control-input--small` : ``)  +
 (alt ? ` au-control-input--alt` : `` ) +
 (block ? ` au-control-input--block` : ``) + 
 (status === 'valid' ? ` au-control-input--valid` : ``)
 (status === 'invalid' ? ` au-control-input--invalid` : `` )
 

<div className={classes} otherProp={otherProp} >
```